pkg/rosa/client.go

```diff
@@ -32,6 +32,23 @@ func NewOCMClient(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (
 	}).Build()
 }
 
+// NewOCMClient2 creates a new OCM client.
+func NewOCMClient2(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (OCMClient, error) {
+	token, url, err := ocmCredentials(ctx, rosaScope)
+	if err != nil {
+		return nil, err
+	}
+	ocmClient, err := ocm.NewClient().Logger(logrus.New()).Config(&ocmcfg.Config{
+		AccessToken: token,
+		URL:         url,
+	}).Build()
```

Comment:
I'm not exactly sure what the difference between this function and `NewOCMClient` is. I see that we wrap the built client in another struct, but the name and godoc string don't really communicate why that's necessary.

Reply:
Ok, so it looks like this is creating a struct that fulfills the new interface, and every call site currently is being replaced with this new function attached to the reconciler.

Is there a reason to make a new function at all? Could we update the logic of the old one? It looks like this PR already phases out use of it.

I don't know if there are external users of the `rosa` package in this repo, but if the concern is about not breaking them, then I'd suggest at least calling this new function something like `NewWrappedOCMClient`.

Reply:
@nrb Are the changes ok with you? Or do you have more question/suggestions?

Reply:
Thanks! And for future reference, the API diff job is mostly so that reviewers are aware of changes; it's not required to merge if the breaking change is acceptable.